### PR TITLE
docs: list all existing debugSession data kinds and their shapes

### PR DESCRIPTION
**Pull request by kpodsiad**
There are 4 debug session data kinds which accepts different `data`. This PR list all possibilities to make things more clear in this area.

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]
Makes sense!

**Review by adpi2 — APPROVED** [review]
Thanks for adding this valuable piece of documentation

File: docs/specification.md [hunk 1]
```diff
@@ -1463,6 +1463,46 @@ export interface DebugSessionParams {
...
+  }
+  ```
+
+  bsp4j classes:
+
+  - [ScalaTestSuites.java](https://github.com/build-server-protocol/build-server-protocol/blob/bdc1558118841d376faba87f83075920d3630886/bsp4j/src/main/xtend-gen/ch/epfl/scala/bsp4j/ScalaTestSuites.java)
+  - [ScalaTestSuiteSelection.java](https://github.com/build-server-protocol/build-server-protocol/blob/bdc1558118841d376faba87f83075920d3630886/bsp4j/src/main/xtend-gen/ch/epfl/scala/bsp4j/ScalaTestSuiteSelection.java)
+
+- `scala-attach-remote`, for which data is an array of [BuildTargetIdentifier](https://github.com/build-server-protocol/build-server-protocol/blob/bdc1558118841d376faba87f83075920d3630886/bsp4j/src/main/xtend-gen/ch/epfl/scala/bsp4j/BuildTargetIdentifier.java).
+
```
Comment (jastice) [inline docs/specification.md:1505]: These data kinda are Scala specific, shouldn't they go in Scala extension doc? We can reference specific docs from here.

**Review by jastice — CHANGES_REQUESTED** [review]

File: docs/specification.md (hunk 1 above)
Comment (kpodsiad) [reply docs/specification.md:1505]: Good point, I'll move them

**Comment by ckipp01** [comment]
@kpodsiad do you care if I rebase this? I can take over and try to get it over the finish line if you'd like.

**Comment by kpodsiad** [comment]
@ckipp01 feel free to pick this up if you want

**Comment by ckipp01** [comment]
Alright, I rebased this and migrated the scala specific stuff to the Scala extension docs and linked to it from the spec. I just realized this is probably important to get in for the work that @baccata is doing in #426

**Review by adpi2 — APPROVED** [review]
LGTM

**Pull request merged**